modules/govcms_ckan_display/plugins/visualisation/bar_chart.inc

```diff
@@ -27,7 +27,23 @@ $plugin = array(
 function govcms_ckan_display_bar_chart_view($file, $display, $config) {
   $element = array();
   $chart_class = 'ckan-bar-chart';
-  $response = govcms_ckan_client_request_records($file->resource_id);
+
+  // Apply filters to the record request.
+  if (!empty($config['ckan_filters']['q'])) {
```

Comment:
If the defaults are added to the plugin settings (at the top of this file) both the $search and $filters checks can be removed and the $config keys will always be present. Same with line & spline plugins (and maybe even the config form).

I know I shouldn't be looking :P but I can't help it, and don't want my baby getting full of unnecessarily IF statements :)